### ISSUE TITLE
Set `:stacks` metadata correctly for polymorphic (generated) instructions

### ISSUE THREAD

**Issue by lspector** [lspector/propeller#12]
Examples showing the problem:

```
(meta (get @instruction-table :float_yank))
=> {:stacks #{:float}}
(meta (get @instruction-table :float_eq))
=> {:stacks #{:float}}
```

The first should include `:integer`, while the second should include `:boolean`.

Until this is fixed, beware that the value of `get-stack-instructions` may include polymorphic instructions that refer to stacks that you do not intend to be using.

I think the idea here, to which the code otherwise adheres (as far as I can see), is that `:stacks` includes all stacks to which the instruction refers or affects.

**Comment by lspector** [comment]
Looked into this for a while and got only as far as suspecting that it may have something to do with macro-expansion and evaluation times in calls to `generate-instructions`. @mcgirjau, perhaps you will see what's going on here if you take a look?

**Comment by lspector** [comment]
Further attempts to fix using macros have failed so far. My Common Lisp-derived thinking about macros isn't cutting it, I guess. I've been able so far to get the correct behavior only by doing it without any macros, which leads to a call to the generator that has to take pairs of functions and function names, which the macro neatly avoided. I'll include my hack that does this below, which I wrote directly in `propeller.push.instructions.polymorphic`; if this sticks it should of course be moved to some generally-accessible utility file, with the `require`s handled in the namespace declarations, etc.

I'd be happy to retain the neater macro-based approach, but if we do that then it has to be fixed for the cases flagged here.

```
(require '(propeller.push.utils.helpers))
(defn make-metadata
  [function stack]
  (->> (:stacks (meta function))
       (replace {:elem (propeller.push.utils.helpers/get-vector-literal-type stack)})
       (cons stack)
       (set)
       (assoc-in (meta function) [:stacks])))

(require '(propeller.push.core))
(defn gen [stacks function-name-pairs]
  (doseq [stack stacks
        [func fname] function-name-pairs]
    (let [instruction-name (keyword (str (name stack) fname))
          metadata (make-metadata func stack)
          new-func (with-meta (partial func stack) metadata)]
      (swap! propeller.push.core/instruction-table assoc instruction-name new-func))))

(gen
  [:boolean :char :code :exec :float :integer :string
   :vector_boolean :vector_float :vector_integer :vector_string]
  [[_dup 'dup]
   [_dup_times '_dup_times]
   [_dup_items '_dup_items]
   [_empty '_empty]
   [_eq '_eq]
   [_flush '_flush]
   [_pop '_pop]
   [_rot '_rot]
   [_shove '_shove]
   [_stack_depth '_stack_depth]
   [_swap '_swap]
   [_yank '_yank]
   [_yank_dup '_yank_dup]])
```

**Comment by lspector** [comment]
Fixed by pull request #13 from mcgirjau/polymorphic-fix
